mesa/examples/advanced/emperor_dilemma/model.py

```diff
@@ -0,0 +1,110 @@
+import random
+
+from agents import EmperorAgent
+
+import mesa
+from mesa.datacollection import DataCollector
+from mesa.space import SingleGrid
+
+
+def compute_compliance(model):
+    # Use model.agents directly
+    if not model.agents:
+        return 0
+    return sum(1 for a in model.agents if a.compliance == 1) / len(model.agents)
+
+
+def compute_enforcement(model):
+    if not model.agents:
+        return 0
+    return sum(1 for a in model.agents if a.enforcement == 1) / len(model.agents)
+
+
+def compute_false_enforcement(model):
+    # Fraction of Disbelievers (-1) who are enforcing the Norm (1)
+    disbelievers = [a for a in model.agents if a.private_belief == -1]
+    if not disbelievers:
+        return 0
+    return sum(1 for a in disbelievers if a.enforcement == 1) / len(disbelievers)
+
+
+class EmperorModel(mesa.Model):
+    def __init__(
```

Comment:
please add docstrings to all methods and classes following the google docstring standard.